File: baekjoon/src/tags/dynamic/No9461.java
package tags.dynamic;

import java.io.*;
import java.util.StringTokenizer;

/**
 * https://www.acmicpc.net/problem/9461
 */
public class No9461 {
    public static final BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
    public static final BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
    public static StringTokenizer tokenizer;

    public static long[] lengths = new long[101];

    public static void setLengths() {
        lengths[1] = lengths[2] = lengths[3] = 1;
        lengths[4] = lengths[5] = 2;
        for (int i = 6; i < lengths.length; i++)
            lengths[i] = lengths[i - 5] + lengths[i - 1];
    }

    public static void main(String[] args) throws IOException {
        setLengths();
        int numberOfInput = Integer.parseInt(reader.readLine());
        for (int i = 0; i < numberOfInput; i++) {
            int number = Integer.parseInt(reader.readLine());
            writer.write(String.valueOf(lengths[number]));
            writer.newLine();
        }
        writer.flush();
    }
}

File: baekjoon/src/tags/dp/No9461.kt
package tags.dp

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

/**
 * https://www.acmicpc.net/problem/9461
 */
fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val testCase = readLine().toInt()
    val dp = LongArray(101)
    dp[1] = 1
    dp[2] = 1
    dp[3] = 1
    dp[4] = 2
    dp[5] = 2
    for (i in 6 until dp.size)
        dp[i] = dp[i - 5] + dp[i - 1]

    with(BufferedWriter(OutputStreamWriter(System.out))) {
        repeat(testCase) {
            val number = readLine().toInt()
            write("${dp[number]}")
            newLine()
        }
        flush()
    }
}